# ATACseqQC

## ATACseqQC installation (miniconda3)
To install ATACseqQC package and implement it on Rstudio:
- Download miniconda3 on linux
  ```{bash}
  mkdir -p ~/miniconda3
  wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda3/miniconda.sh
  bash ~/miniconda3/miniconda.sh -b -u -p /Storage1/fs1/yeli/Active/l.ronghan/miniconda3
  rm ~/miniconda3/miniconda.sh
  ```
- Add conda to R libPath
  Open ~/.Rprofile:
  ```{R}
  lib_path <- paste0('/storage1/fs1/yeli/Active/l.ronghan/R_libraries/', R.version$major, ".", R.version$minor)
  
  for (devlib in lib_path){
    if (!file.exists(devlib))
    dir.create(devlib, recursive = TRUE)
    x <- .libPaths()
    x <- .libPaths(c(devlib,x))
  }
  
  .libPaths(c(.libPaths(), '/storage1/fs1/yeli/Active/l.ronghan/miniconda3/lib/R/library'))
  ```
- Install ATACseqQC through miniconda3
  Open Rstudio, make sure that the miniconda3 is existing in the R `.libPaths()`.
  Install ATACseqQC through this command line:
  ```{bash}
  conda install bioconda::bioconductor-atacseqqc
  ```
  
## ATACseqQC pipeline
### Library installation
```{R}
setwd("/storage1/fs1/yeli/Active/l.ronghan/projects/1.bulkATAC_0901")
library(BiocManager)
BiocManager::install(c("ChIPpeakAnno", "MotifDb", "GenomicAlignments",
           "BSgenome.Hsapiens.UCSC.hg19", "TxDb.Hsapiens.UCSC.hg19.knownGene",
           "phastCons100way.UCSC.hg19"))
```

### input the bamFile from the ATACseqQC package 
```{R}
bamfilr <- "filtered_bam_files/LIB047013-DIL01_22CYYHLT4_S747_L007.bam"
bamfile.labels <- gsub(".bam", "", basename(bamfile))
```

### Estimate the library complexity
```{R}
estimateLibComplexity(readsDupFreq(bamfile))
```
### Fragment size distribution
```{R}
## generate fragement size distribution
fragSize <- fragSizeDist(bamfile, bamfile.labels)
```
### Nucleosome positioning
#### Adjust the read start sites
```{bash}
bash 0.data_tn5_shifting
```
