# ATACseqQC

## ATACseqQC installation (miniconda3)
To install ATACseqQC package and implement it on Rstudio:
- Download miniconda3 on linux
  ```{bash}
  mkdir -p ~/miniconda3
  wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda3/miniconda.sh
  bash ~/miniconda3/miniconda.sh -b -u -p /Storage1/fs1/yeli/Active/l.ronghan/miniconda3
  rm ~/miniconda3/miniconda.sh
  ```
- Add conda to R libPath
  Open ~/.Rprofile:
  ```{R}
  lib_path <- paste0('/storage1/fs1/yeli/Active/l.ronghan/R_libraries/', R.version$major, ".", R.version$minor)
  
  for (devlib in lib_path){
    if (!file.exists(devlib))
    dir.create(devlib, recursive = TRUE)
    x <- .libPaths()
    x <- .libPaths(c(devlib,x))
  }
  
  .libPaths(c(.libPaths(), '/storage1/fs1/yeli/Active/l.ronghan/miniconda3/lib/R/library'))
  ```
- Install ATACseqQC through miniconda3
  Open Rstudio, make sure that the miniconda3 is existing in the R `.libPaths()`.
  Install ATACseqQC through this command line:
  ```{bash}
  conda install bioconda::bioconductor-atacseqqc
  ```
  
## ATACseqQC pipeline
### Library installation
```{R}
setwd("/storage1/fs1/yeli/Active/l.ronghan/projects/1.bulkATAC_0901")
library(BiocManager)
BiocManager::install(c("ChIPpeakAnno", "MotifDb", "GenomicAlignments",
           "BSgenome.Hsapiens.UCSC.hg19", "TxDb.Hsapiens.UCSC.hg19.knownGene",
           "phastCons100way.UCSC.hg19"))
```

### input the bamFile from the ATACseqQC package 
```{R}
bamfilr <- "filtered_bam_files/LIB047013-DIL01_22CYYHLT4_S747_L007.bam"
bamfile.labels <- gsub(".bam", "", basename(bamfile))
```

### Estimate the library complexity
```{R}
estimateLibComplexity(readsDupFreq(bamfile))
```
### Fragment size distribution
```{R}
## generate fragement size distribution
fragSize <- fragSizeDist(bamfile, bamfile.labels)
```
### Nucleosome positioning
#### Adjust the read start sites

Tn5 transposase has been shown to bind as a dimer and inserts two adaptors into accessible DNA locations separated by 9 bp.

Therefore, for downstream analysis, such as peak-calling and footprinting, all reads in input bamfile need to be shifted. The function shiftGAlignmentsList can be used to shift the reads. By default, all reads aligning to the positive strand are offset by +4bp, and all reads aligning to the negative strand are offset by -5bp.

The method ATACseqQC uses for Tn5 shifting occupies a lot of CPU memory, so we use ENCODE pipeline for Tn5 shifting (`awk`).
```{bash}
bash 0.data_tn5_shifting
```
After running this script, we will get the tagAlin objects (.bed.gz). We need to transform them into the GAlinment objects for downstreaming analysis:

```{R}
tagalign_file <- "sample1.tn5.tagAlign.bed"  # 替换为你的文件名
tag_data <- read.table(tagalign_file, header = FALSE, sep = "\t", stringsAsFactors = FALSE)

gal1 <- GAlignments(
  seqnames = Rle(tag_data[, 1]),         # chromosome
  pos = tag_data[, 2] + 1,               # start
  strand = Rle(tag_data[, 6]),            # strand
)

print(gal1)
```
#### Promoter/Transcript body (PT) score
PT score is calculated as the coverage of promoter divided by the coverage of its transcript body. PT score will show if the signal is enriched in promoters.
```{R}
gal1 <- path/to/shiftedfiles

library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txs <- transcripts(TxDb.Hsapiens.UCSC.hg19.knownGene)

pt <- PTscore(gal1, txs)
plot(pt$log2meanCoverage, pt$PT_score, 
     xlab="log2 mean coverage",
     ylab="Promoter vs Transcript")
```

#### Nucleosome Free Regions (NFR) score
NFR score is a ratio between cut signal adjacent to TSS and that flanking the corresponding TSS.
The NFR score for each TSS is calculated as NFR-score = log2(nf) - log2((n1+n2)/2). 
```{R}
nfr <- NFRscore(gal1, txs)
plot(nfr$log2meanCoverage, nfr$NFR_score, 
     xlab="log2 mean coverage",
     ylab="Nucleosome Free Regions score",
     main="NFRscore for 200bp flanking TSSs",
     xlim=c(-10, 0), ylim=c(-5, 5))
```
