---
title: "differential_analysis"
author: "Ronghan Li"
date: "2024-10-23"
output: html_document
---
```{r}
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)  # Use the appropriate genome annotation for your species
library(org.Hs.eg.db)  # Human-specific, replace if using another species
```

```{r}
count <- read.table("/Users/ronghanli/Desktop/filtered_peaks_files/countmatrix.tsv", header=TRUE, row.names=1)
count
```
```{r}
colnames(count)=c("Chr","Start","End","Strand","control1_005", "control2_007", "treat1_002", "treat2_004")
count_clean <- count[, c("Chr", "Start", "End", "control1_005", "control2_007", "treat1_002", "treat2_004")]
colnames(count_clean)[1:3] <- c("seqnames", "start", "end")
count_clean
```
```{r}
library(GenomicRanges)
```

```{r}
gr <- GRanges(seqnames = count$Chr,ranges = IRanges(start = count$Start, end = count$End))

peak_annotation <- annotatePeak(gr, tssRegion=c(-3000, 3000),
TxDb=TxDb.Hsapiens.UCSC.hg38.knownGene,
annoDb="org.Hs.eg.db")

peak_annotation_df <- as.data.frame(peak_annotation)
```
```{r}
merged_df <- merge(count_clean, peak_annotation_df, by=c("seqnames", "start", "end"))
write.table(merged_df,"/Users/ronghanli/Desktop/filtered_peaks_files/countmatrix_cleaned.tsv",sep = "\t", row.names = FALSE, quote = FALSE)
```


```{r}
library(dplyr)
```

```{r}
summarized_counts <- merged_df %>%
group_by(SYMBOL) %>%
summarize(across(starts_with("control") | starts_with("treat"), sum))
```

```{r}

```

```{r}
#filter for promoter region

promoter_df <- merged_df %>%
filter(annotation == "Promoter (<=1kb)")
#then sum by gene

summarized_counts_promoter <- promoter_df %>%
group_by(SYMBOL) %>%
summarize(across(starts_with("control") | starts_with("treat"), sum))
summarized_counts_promoter
```


```{r}
write.table(summarized_counts_promoter,"/Users/ronghanli/Desktop/filtered_peaks_files/summarized_counts_promoter.tsv",sep = "\t", row.names = FALSE, quote = FALSE)
```

```{r}
sample_info <- data.frame( condition = c("control", "control", "treat", "treat"), row.names = c("control1_005", "control2_007", "treat1_002", "treat2_004"))
sample_info
```

```{r}
cleaned_df <- summarized_counts_promoter[!(is.na(summarized_counts_promoter$SYMBOL) | summarized_counts_promoter$SYMBOL == ""), ]

write.table(cleaned_df, "/Users/ronghanli/Desktop/filtered_peaks_files/summarized_counts_promoter.tsv", 
            sep = "\t", row.names = FALSE, quote = FALSE)
```


```{r}
summarized_counts <- read.table("/Users/ronghanli/Desktop/filtered_peaks_files/summarized_counts_promoter.tsv",
header = TRUE,      # The file has a header
sep = "\t",
row.names = 1 # Tab-separated values
)      # First column (SYMBOL) as row names

summarized_counts
```
```{r}
library(DESeq2)
dds <- DESeqDataSetFromMatrix(countData=summarized_counts, colData=sample_info, design=~condition)
dds
```

```{r}
vsd<-vst(dds,blind=FALSE)
vsd
```
```{r}
normalization <- assay(vst(dds, blind = FALSE))
```

```{r}
write.csv(normalization, '/Users/ronghanli/Desktop/filtered_peaks_files/norm_matrix_promoter.csv')
```

```{r}
write.table(normalization, file = "/Users/ronghanli/Desktop/filtered_peaks_files/norm_matrix_promoter.txt", sep = "\t", row.names = TRUE, quote = FALSE)
```

```{r}
dds1 <- estimateSizeFactors(dds)
dds1
```
```{r}
normalized_counts <- counts(dds1, normalized = TRUE)
```

```{r}
dds <- DESeq(dds)
```


```{r}
res <- results(dds, contrast = c("condition", "treat", "control"))
res
```
```{r}
write.csv(res,file="/Users/ronghanli/Desktop/filtered_peaks_files/DEG_promoter.csv")
```

```{r}
library(EnhancedVolcano)
library(ggplot2)
volcano_plot<-EnhancedVolcano(res,
  lab = rownames(res),
  x = 'log2FoldChange',
  y = 'pvalue',
  pCutoff = 0.05,
  FCcutoff = 0.5,
  pointSize = 3,
  labSize = 2,
  labCol = "black",
  title = "Volcano Plot of DEGs",
  subtitle = "DESeq2 Analysis",
  ylim = c(0, 10),
  caption = "p-value < 0.05 and log2FC > 2 are highlighted",
  xlim = c(-2.5, 2.5))+
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold"),
    plot.subtitle = element_text(hjust = 0.5)
)


ggsave("volcano_plot.png", volcano_plot, width = 8, height = 14)
```





```{r}
volcano_plot
```


```{r}
library(DESeq2)
library(pheatmap)
```

```{r}
res
```

```{r}
res_ordered <- res[order(res$log2FoldChange, decreasing = TRUE), ]
res_ordered
```
```{r}
top20_upregulated <- head(res_ordered, 20)
top20_upregulated
```
```{r}
top20_downregulated <- tail(res_ordered, 20)
top20_downregulated
```

```{r}
top_genes <- top20_upregulated
top_gene_names <- rownames(top_genes)
top_gene_names
```

```{r}
normalized_counts <- counts(dds, normalized = TRUE)
top_gene_counts <- normalized_counts[top_gene_names, ]
top_gene_counts
```

```{r}
library(RColorBrewer)

color_palette <- colorRampPalette(rev(brewer.pal(n = 11, name = "RdBu")))(100)
log_transformed_counts <- log2(top_gene_counts + 1)
top20_heatmap <- pheatmap(log_transformed_counts,
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  show_rownames = TRUE,
  color = color_palette, # sample_info contains condition data
  scale = "row", # scale by row to show relative changes
  fontsize_row = 10,
  main = "Heatmap of Top 20 Upregulated Genes")

ggsave("top20_heatmap.png", top20_heatmap, width = 8, height = 14)
```

```{r}
hallmark_genes <- c("ABCA1", "ABI1", "ACVR1B", "ACVR2A", "ADM", "ADORA2B",
"ADRM1", "AHR", "APLNR", "AQP9", "ATP2A2", "ATP2B1",
"ATP2C1", "AXL", "BDKRB1", "BEST1", "BST2", "BTG2",
"C3AR1", "C5AR1", "CALCRL", "CCL17", "CCL2", "CCL20",
"CCL22", "CCL24", "CCL5", "CCL7", "CCR7", "CCRL2",
"CD14", "CD40", "CD48", "CD55", "CD69", "CD70",
"CD82", "CDKN1A", "CHST2", "CLEC5A", "CMKLR1",
"CSF1", "CSF3", "CSF3R", "CX3CL1", "CXCL10",
"CXCL11", "CXCL6", "CXCL9", "CXCR6", "CYBB",
"DCBLD2", "EBI3", "EDN1", "EIF2AK2", "EMP3",
"ADGRE1", "EREG", "F3", "FFAR2", "FPR1", "FZD5",
"GABBR1", "GCH1", "GNA15", "GNAI3", "GP1BA",
"GPC3", "GPR132", "GPR183", "HAS2", "HBEGF",
"HIF1A", "HPN", "HRH1", "ICAM1", "ICAM4", "ICOSLG",
"IFITM1", "IFNAR1", "IFNGR2", "IL10", "IL10RA",
"IL12B", "IL15", "IL15RA", "IL18", "IL18R1",
"IL18RAP", "IL1A", "IL1B", "IL1R1", "IL2RB",
"IL4R", "IL6", "IL7R", "CXCL8", "INHBA", "IRAK2",
"IRF1", "IRF7", "ITGA5", "ITGB3", "ITGB8", "KCNA3",
"KCNJ2", "KCNMB2", "KIF1B", "KLF6", "LAMP3",
"LCK", "LCP2", "LDLR", "LIF", "LPAR1", "LTA",
"LY6E", "LYN", "MARCO", "MEFV", "MEP1A", "MET",
"MMP14", "MSR1", "MXD1", "MYC", "NAMPT", "NDP",
"NFKB1", "NFKBIA", "NLRP3", "NMI", "NMUR1",
"NOD2", "NPFFR2", "OLR1", "OPRK1", "OSM",
"OSMR", "P2RX4", "P2RX7", "P2RY2", "PCDH7",
"PDE4B", "PDPN", "PIK3R5", "PLAUR", "PROK2",
"PSEN1", "PTAFR", "PTGER2", "PTGER4", "PTGIR",
"PTPRE", "PVR", "RAF1", "RASGRP1", "RELA",
"RGS1", "RGS16", "RHOG", "RIPK2", "RNF144B",
"ROS1", "RTP4", "SCARF1", "SCN1B", "SELE",
"SELL", "SELENOS", "SEMA4D", "SERPINE1",
"SGMS2", "SLAMF1", "SLC11A2", "SLC1A2",
"SLC28A2", "SLC31A1", "SLC31A2", "SLC4A4",
"SLC7A1", "SLC7A2", "SPHK1", "SRI", "STAB1",
"TACR1", "TACR3", "TAPBP", "TIMP1", "TLR1",
"TLR2", "TLR3", "TNFAIP6", "TNFRSF1B",
"TNFRSF9", "TNFSF10", "TNFSF15", "TNFSF9",
"TPBG", "VIP")
```

```{r}
normalized_counts <- counts(dds, normalized = TRUE)
existing_hallmark_genes <- hallmark_genes[hallmark_genes %in% rownames(normalized_counts)]
hallmark_counts <- normalized_counts[existing_hallmark_genes, ]
hallmark_counts
```

```{r}
res_hallmark <- res[rownames(res) %in% existing_hallmark_genes, ]
res_hallmark
```
```{r}
res_ordered <- res_hallmark[order(res_hallmark$log2FoldChange, decreasing = TRUE), ]
res_ordered
```

```{r}
top20_hallmark_upregulated <- head(res_ordered, 20)
top20_hallmark_upregulated
```
```{r}
top_hallmark_genes <- top20_hallmark_upregulated
top_hallmark_gene_names <- rownames(top_hallmark_genes)
top_hallmark_gene_names
```

```{r}
top_hallmark_gene_counts <- normalized_counts[top_hallmark_gene_names, ]
top_hallmark_gene_counts
```
```{r}
sample_info
```

```{r}
log_hallmark_transformed_counts <- log2(top_hallmark_gene_counts + 1)
color_palette <- colorRampPalette(rev(brewer.pal(n = 11, name = "RdBu")))(100)
sample_info$condition <- factor(sample_info$condition, levels = c("control1_005_final.bam", "control2_007_final.bam", "treat1_002_final.bam", "treat2_004_final.bam"))
annotation <- data.frame(Condition = c("Vehicle", "Vehicle", "rGAL", "rGAL"))
rownames(annotation) <- colnames(log_hallmark_transformed_counts)

hallmark_heatmap <- pheatmap(log_hallmark_transformed_counts,
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  show_rownames = TRUE,
  annotation_col = annotation,  # Use the custom annotation
  scale = "row",  # Scale by row to show relative changes
  fontsize_row = 10,
  main = "Heatmap of Top 20 Hallmark Inflammatory Response Genes",
  color = color_palette,  # Use the custom color palette
  cellwidth = 15,
  cellheight = 10)  

ggsave("top20_hallmark_heatmap.png", hallmark_heatmap, width = 8, height = 14)
```

```{r}
inflam_heatmap<- pheatmap(log_hallmark_transformed_counts,
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  show_rownames = TRUE,
  annotation_col = annotation,  # use the new annotation with control/treatment
  scale = "row", # scale by row to show relative changes
  fontsize_row = 10,
  main = "Heatmap of Top 20 Upregulated and Downregulated Hallmark Inflammatory Response Genes",
  color = color_palette,  # Use the custom color palette
  cellwidth = 15, cellheight = 10)

ggsave("top20_inflam_heatmap.png", inflam_heatmap, width = 8, height = 14)
```


